pkg/parquetquery/iters.go

```diff
@@ -30,14 +33,14 @@ import (
 //	    D      0,  1,  0
 //	  E        0,  2, -1
 //
-// Currently supports 6 levels of nesting which should be enough for anybody. :)
-type RowNumber [6]int32
+// Currently supports 8 levels of nesting which should be enough for anybody. :)
+type RowNumber [8]int32
```

Comment:
the perf hit for extending to 8 is not great. can we get two quick cpu profiles. one from main and one from this branch to identify and reduce the impact.